lib/steep/project/completion_provider.rb

```diff
@@ -293,6 +294,10 @@ def index_for(string, line:, column:)
 
         index
       end
+
+      def inherited_method?(method, type)
+        method.implemented_in.name&.name != type.name.name
```

Comment:
I think we should omit the last `.name`.

```suggestion
        method.implemented_in.name != type.name
```

Reply:
Ah, this doesn't seem to work right now.

It looks like `method.implemented_in.name.namespace` returns an `RBS::Namespace` and `type.name.namespace` returns a `Steep::AST::Namespace` so the equality operator in `Steep::Names::Base` fails since it relies on the namespaces being equal. 

We could change the equality operator for `Steep::AST::Namespace`? Not sure what relies on that equality or if the `CompletionProvider` is more special and it's the only place where we're comparing `RBS::Namespace`s to `Steep::AST::Namespace`.

Reply:
Ah, I got it. 💦